# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../etcnet/etcnet-0.8.5-tap-support.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Submitted By: Christian Wiese <morfoh at opensde dot org>
Date: 2007-03-01
Initial Package Version: 0.8.5
Upstream Status: submitted
Origin: Christian Wiese <morfoh at opensde dot org>
        http://trac.opensde.org/browser/package/trunk/network/etcnet/
Description: This patch adds tap interface support to etcnet using the tunctl
             command.

diff -ruN etcnet-0.8.5/etc/net/options.d/00-default etcnet-0.8.5-tap/etc/net/options.d/00-default
--- etcnet-0.8.5/etc/net/options.d/00-default	2006-09-23 20:45:27.000000000 +0300
+++ etcnet-0.8.5-tap/etc/net/options.d/00-default	2007-03-01 18:00:31.000000000 +0200
@@ -63,6 +63,6 @@
 
 IFGROUP[0]='lo dummy'
 IFGROUP[1]='eth plip usb dvb bnep'
-IFGROUP[2]='vlan bond bri teql'
+IFGROUP[2]='vlan bond bri teql tap'
 IFGROUP[3]='iptun ipsectun'
 IFGROUP[4]='ppp'
diff -ruN etcnet-0.8.5/etc/net/scripts/create-tap etcnet-0.8.5-tap/etc/net/scripts/create-tap
--- etcnet-0.8.5/etc/net/scripts/create-tap	1970-01-01 02:00:00.000000000 +0200
+++ etcnet-0.8.5-tap/etc/net/scripts/create-tap	2007-03-01 17:57:52.000000000 +0200
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+pickup_defaults
+pickup_options
+
+$MODPROBE tun || {
+        print_error "tun kernel module not found"
+        exit 2
+}
+
+[ -x "${TUNCTL:=$DEFAULT_TUNCTL}" ] || {
+        print_error "$TUNCTL does not exist or is not executable. Try installing tunctctl package on your system."
+        exit 1
+}
+
+$TUNCTL -t $NAME
diff -ruN etcnet-0.8.5/etc/net/scripts/destroy-tap etcnet-0.8.5-tap/etc/net/scripts/destroy-tap
--- etcnet-0.8.5/etc/net/scripts/destroy-tap	1970-01-01 02:00:00.000000000 +0200
+++ etcnet-0.8.5-tap/etc/net/scripts/destroy-tap	2007-03-01 17:57:52.000000000 +0200
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+pickup_defaults
+pickup_options
+
+[ -x "${TUNCTL:=$DEFAULT_TUNCTL}" ] || {
+	print_error "$TUNCTL does not exist or is not executable. Try installing the tunctl package on your system."
+	exit 1
+}
+
+$TUNCTL -d $NAME
+try_rmmod tun
diff -ruN etcnet-0.8.5/etc/net/scripts/functions etcnet-0.8.5-tap/etc/net/scripts/functions
--- etcnet-0.8.5/etc/net/scripts/functions	2007-01-28 14:27:19.000000000 +0200
+++ etcnet-0.8.5-tap/etc/net/scripts/functions	2007-03-01 17:59:48.000000000 +0200
@@ -41,6 +41,7 @@
 DEFAULT_SZAP_ARGS='-n 1 -x'
 DEFAULT_DVBNET=/usr/bin/dvbnet
 DEFAULT_PAND=/usr/bin/pand
+DEFAULT_TUNCTL=/usr/sbin/tunctl
 
 trim()
 {
@@ -487,3 +488,4 @@
 declare -rx DEFAULT_PPPINITCHAT DEFAULT_PPPCONNECTCHAT DEFAULT_PPPDISCONNECTCHAT
 declare -rx DEFAULT_CHATOPTIONS DEFAULT_PPPOE DEFAULT_PPPOEOPTIONS DEFAULT_PPTP
 declare -rx DEFAULT_SZAP DEFAULT_SZAP_ARGS DEFAULT_DVBNET DEFAULT_PAND
+declare -rx DEFAULT_TUNCTL
